operator/pkg/gateway-api/helpers.go

```diff
@@ -178,3 +179,22 @@ func setMergedLabelsAndAnnotations(temp, desired client.Object) {
 	temp.SetAnnotations(mergeMap(temp.GetAnnotations(), desired.GetAnnotations()))
 	temp.SetLabels(mergeMap(temp.GetLabels(), desired.GetLabels()))
 }
+
+// hasCiliumManagedGateway checks if any of the route's parent refs point to a Cilium-managed gateway.
+// Returns true if at least one parent gateway is managed by Cilium, false otherwise.
+func hasCiliumManagedGateway(ctx context.Context, c client.Client, parentRefs []gatewayv1.ParentReference, getGatewayFn func(ref gatewayv1.ParentReference) (*gatewayv1.Gateway, error), logger *slog.Logger) (bool, error) {
+	for _, parent := range parentRefs {
+		if (parent.Group == nil || *parent.Group == gatewayv1.GroupName) &&
```

Comment:
This check exists in the helpers module, but when I tried to import/reference that method here I got some mod-vendor compilation errors that I didn't have enough golang-fu to resolve, so I've inlined them for now.